CASMTRIAGE-8315: Updating firmware on m001 doc release/1.7 (#6207)
Cray-HPE/docs-csm: operations/firmware/Updating_Firmware_m001.md

    - `ipaddressOfM001` = IP address of `ncn-m001` node
    - `filename` = Filename of the downloaded image

    ```bash
    curl -k -u ${USERNAME}:${BMC_PASSWORD} https://${BMC_ADDRESS}/redfish/v1/UpdateService/Actions/SimpleUpdate -H 'Content-Type: application/json' \
        -d '{"ImageURI":"http://ipaddressOfM001:8770/filename", "TransferProtocol":"HTTP", "UpdateComponent":"BIOS"}'
    ```

    Example output:

     ```json
     {"@odata.type":"#UpdateService.v1_5_0.UpdateService","Messages":[{"@odata.type":"#Message.v1_0_7.Message","Message":"A new task /redfish/v1/TaskService/Tasks/596 was created.","MessageArgs":["/redfish/v1/TaskService/Tasks/596"],"MessageId":"Task.1.0.New","Resolution":"None","Severity":"OK"},{"@odata.type":"#Message.v1_0_7.Message","Message":"Device is prepareing flash firmware for action SimpleUpdate.","MessageArgs":["SimpleUpdate"],"MessageId":"UpdateService.1.0.PrepareUpdate","Resolution":"None","Severity":"OK"}]}
     ```

1. Verify successful update.

     When updating the Gigabyte BIOS, the BIOS version number will not change until the node is rebooted.
     To verify a successful update, check the task that was created.

>      1. Using the return json packet, find the line resembling the following.

Reviewer: ```suggestion
     1. In the JSON response from the previous step, find the text that resembles the following.
```

     Using the task number (596 in the above example), check the state of the task:

     ```bash
    curl -sk -u ${USERNAME}:${BMC_PASSWORD} https://${BMC_ADDRESS}/redfish/v1/TaskService/Tasks/596 | jq .TaskState

>      ```

Reviewer: Indentation is misaligned.

````suggestion
     curl -sk -u ${USERNAME}:${BMC_PASSWORD} https://${BMC_ADDRESS}/redfish/v1/TaskService/Tasks/596 | jq .TaskState
     ```
````

    1. Click `"Flash"`.

    > After updating System ROM (BIOS), `ncn-m001` will need to be rebooted. Follow the [Reboot NCNs](../node_management/Reboot_NCNs.md) procedure to reboot `ncn-m001`.

### Flash HPE `ncn-m001` using `ilorest`

1. (`ncn-m001#`) Install `ilorest` rpm on `ncn-m001`.

    ```bash
    zypper install ilorest
    ```

1. {`ncn-m001#`) Check firmware versions before making changes on `ncn-m001`.

    ```bash
    ilorest serverinfo  |grep "Firmware:" -A3

>     Firmware: 
>     ------------------------------------------------
>     iLO 5 : 3.02 Feb 22 2024
>     System ROM : A43 v3.60 (01/21/2025)
>     ```

Reviewer: ````suggestion
    ```

    Example output:

    ```text
    Firmware: 
    ------------------------------------------------
    iLO 5 : 3.02 Feb 22 2024
    System ROM : A43 v3.60 (01/21/2025)
    ```
````

1. {`ncn-m001#`) Update BIOS on `ncn-m001` using the downloaded System ROM file.

    ```bash
    ilorest uploadcomp --component=A43_3.70_03_21_2025.signed.flash --update_target
    sleep 90
    ```

1. {`ncn-m001#`) Update BMC on `ncn-m001` using the downloaded iLO file.

    ```bash
    ilorest uploadcomp --component=ilo5_311.bin --forceupload --update_target
    sleep 60
    ```

1. {`ncn-m001#`) Using `ipmitool`, login to the `ncn-m001` console, using the correct user, password, and BMC IP address so that you can watch the console log as the node boots

    ```bash
    ipmitool -I lanplus -U $USERNAME -P $BMC_PASSWORD -H $BMC_ADDRESS sol activate

>     ncn-m001 login: root

Reviewer: ````suggestion
    ```

    Example console:

    ```console
    ncn-m001 login: root
````

    Password:
    ```

1. {`ncn-m001#`) Reboot `ncn-m001`. Do not accidentally issue this command on another node!

    ```bash
    shutdown -r now
    ```

1. {`ncn-m001#`) Once the node is back up check firmware versions.

    ```bash
    ilorest serverinfo  |grep "Firmware:" -A3

>     Firmware: 

Reviewer: ````suggestion
    ```

    Example output:

    ```text
    Firmware: 
````

    1. Click `"Flash"`.

    > After updating System ROM (BIOS), `ncn-m001` will need to be rebooted. Follow the [Reboot NCNs](../node_management/Reboot_NCNs.md) procedure to reboot `ncn-m001`.

### Flash HPE `ncn-m001` using `ilorest`

> 1. (`ncn-m001#`) Install `ilorest` rpm on `ncn-m001`.

Reviewer: ```suggestion
1. (`ncn-m001#`) Install `ilorest` RPM on `ncn-m001`.
```
